src/foundry/foundry.js/hooks.d.ts

```diff
@@ -1,830 +1,804 @@
+     * @remarks This is called when the user mouse is entering or leaving a hover state over a {@link PlaceableObject}.
+     * @param object - the PlaceableObject
+     * @param hover  - whether the mouse is hovering over the PlaceableObject
+     * @typeParam P  - the type of the PlaceableObject
+     * @remarks The name for this hook is dynamically created by joining 'hover' and the type name of the PlaceableObject.
      * @remarks This is called by {@link Hooks.callAll}.
-     * @see {@link Game#setupGame}
+     * @see {@link PlaceableObject#_onHoverIn}
+     * @see {@link PlaceableObject#_onHoverOut}
      */
-    ready: () => unknown;
+    type HoverPlaceableObject<P extends PlaceableObject = PlaceableObject> = (object: P, hover: boolean) => unknown;
 
     /**
-     * This is called as last step when rendering a {@link ChatMessage}.
-     * @param message     - the ChatMessage
-     * @param jq          - the JQuery of the rendered ChatMessage
-     * @param messageData - the data of the message
-     * @remarks This is called by {@link Hooks.callAll}.
-     * @see {@link ChatMessage#render}
+     * @remarks This is called after copying {@link PlaceableObject}s in a copy-paste action, but before embedding them
+     * into the {@link PlaceablesLayer}.
+     * @param copied - the originally copied PlaceableObjects
+     * @param pasted - the pasted copies with new coordinates
+     * @param P      - the type of the PlaceableObject
+     * @remarks The name for this hook is dynamically created by joining 'paste' with the type name of the
+     * PlaceableObject.
+     * @remarks This is called by {@link Hooks.call}.
+     * @see {@link PlaceablesLayer#pasteObjects}
      */
-    renderChatMessage: (message: ChatMessage, jq: JQuery, messageData: ChatMessage.MessageData) => unknown;
+    type PastePlaceableObject<P extends PlaceableObject = PlaceableObject> = (
+      copied: P[],
+      pasted: P[]
+    ) => boolean | void;
 
     /**
-     * This is called after {@link AVSettings} are changed.
-     * @param settings - the AVSettings
-     * @param changed  - an object reflecting the changed settings
-     * @remarks This is called by {@link Hooks.callAll}.
-     * @see {@link AVSettings#_onSettingsChanged}
+     * @remarks This is called after copying {@link Wall}s in a copy-paste action, but before embedding them into the
+     * {@link WallsLayer}.
+     * @param copied - the originally copied Walls
+     * @param pasted - the pasted copies with new coordiantes
+     * @param W      - the type of the Wall
+     * @remarks The name for this hook is dynamically created by joining 'paste' with the type name of the Wall.
+     * @remarks This is called by {@link Hooks.call}.
+     * @see {@link WallsLayer#pasteObjects}
      */
-    rtcSettingsChanged: (settings: DeepPartial<AVSettings.Settings>, changed: object) => unknown;
+    type PasteWall<W extends Wall = Wall> = (copied: W[], pasted: W[]) => boolean | void;
```

Comment:
I think this is just a special case of `PastePlaceableObject`, isn't it? Do we actually need it separately?